java/ql/lib/ext/java.io.model.yml

```diff
@@ -3,11 +3,22 @@ extensions:
       pack: codeql/java-all
       extensible: sinkModel
     data:
+      - ["java.io", "File", True, "canRead", "()", "", "Argument[this]", "path-injection", "manual"]
+      - ["java.io", "File", True, "canWrite", "()", "", "Argument[this]", "path-injection", "manual"]
```

Comment:
> why these and not `canExecute`? (I'm sure there's a good reason that I'm missing.)

I'm sorry to say the reason is that I accidentally didn't add it 😅. I'll add it now.

> And if we're doing ones that are equivalent to `exists`, how about `isFile`?

Since we want to be consistent, we indeed should add `isFile`, `isDirectory`, and `isHidden`. I guess I initially didn't because it feels somewhat wrong due to their low impact, and that these calls could precisely be part of security checks. But it's the same discussion as with `exists`, and we already decided to model it, so let's add these as well.